sentry-core/src/main/java/io/sentry/core/Hub.java

```diff
@@ -50,7 +49,7 @@ public boolean isEnabled() {
   public SentryId captureEvent(SentryEvent event) {
     SentryId sentryId;
     StackItem item = stack.peek();
-    sentryId = item.client.captureEvent(event, item.scope);
+    sentryId = item != null ? item.client.captureEvent(event, item.scope) : SentryId.EMPTY_ID;
```

Comment:
`assert item != null` would or something that works on android would be better.
in other words, if item is null, we have a bug
We could do a `log(SentryLevel.FATAL)` in this case

Reply:
I'd not use `assert` on public methods, I'd log the `SentryEvent` if `StackItem` is null though, does it sound good?
same for the other cases.

Reply:
`assert` doesn't get compiled in `release` builds so wouldn't matter. But Android complains about using `assert` anyway.

If StackItem is null the SDK has a huge bug because this class guards all access to it and it cannot let it remove the root scope. Please review the code and see if you spot a way where that could happen.

Logging `FATAL` StackItem is null would be ideal so we can get it fixed. If the SDK is buggy, logging SentryEvent doesn't add much value. These logs are to diagnose problems with the SDK and not let the user aware of errors we captured.